src/app/utils.ts

```diff
@@ -212,3 +212,18 @@ export const onEscHandler = (element: React.RefObject<HTMLElement>, isEnabled: b
 export const useOnEsc = (element: React.RefObject<HTMLElement>, isEnabled: boolean, cb: () => void) => {
   React.useEffect(onEscHandler(element, isEnabled, cb), [element, isEnabled]);
 };
+
+export const makeApiCallOrThrow = async<T>(promise: Promise<T>): Promise<T> => {
+  return promise.catch((error) => {
+    if (error.status !== 422) {
+      throw new Error(error);
```

Comment:
i feel like you're this is going to potentially nest things that shouldn't be nested. maybe doing 
```
return Promise.reject(error);
```

would be better? i'm assuming what you're being passed in this handler is already an error, so nesting it in another error object might not be ideal